jscomp/syntax/src/reactjs_jsx_v4.ml

```diff
@@ -369,7 +369,16 @@ let transformUppercaseCall3 ~config modulePath mapper jsxExprLoc callExprLoc
     recursivelyTransformedArgsForMake
     @
     match childrenExpr with
-    | Exact children -> [(labelled "children", children)]
+    | Exact children -> (
+      match config.React_jsx_common.mode with
+      | "automatic" -> [(labelled "children", children)]
+      | _ ->
```

Comment:
Perhaps it's obvious by the discussion, but this is the aspect I'd like to dig into a little bit.
Can you explain again why there are 2 apparently different treatments for automatic vs non-automatic?

Is is possible to explain the need for a difference at a conceptual level, or is this just due to some implementation details? I wonder especially if this reflects a difference in the way React already treats the different modes, or if it is specific to ReScript.

Also why is that a string? Not clear we catch all the cases here, especially if more cases will be added later in addition to the current 3 cases.

Reply:
> I wonder especially if this reflects a difference in the way React already treats the different modes

That's right, React handles the representation of classic and automatic differently with different runtimes. This is easy to see in the Babel playground.
<img width="2138" alt="image" src="https://user-images.githubusercontent.com/36963424/236096488-7f61165b-7158-4589-bbdc-ab7242520a8a.png">
(I couldn't find the share function in the Babel playground, so here's a screenshot.)

> Also why is that a string? Not clear we catch all the cases here, especially if more cases will be added later in addition to the current 3 cases.

It doesn't have to be a string, because we take a string as an argument to -bs-jsx-mode, but it could be parsed as a variant. As for the exception handling in the parsing, I think you could use the default value and not have to use the option type.

Reply:
Suddenly, I'm wondering if ReScript should only support the automatic React runtime, since the classic runtime is for React v17 and earlier.